mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -191,6 +191,8 @@ public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
 			for ( int i = 0; i < searchObjects.length; i++ ) {
 				if ( dbObject.get( ID_FIELDNAME ).equals( searchObjects[i] ) ) {
 					tuples.set( i, createTuple( keys[i], tupleContext, dbObject ) );
```

Comment:
Do we know for sure it's limited to "up to a 1000" or is that just the expected use case?

I like your idea. Hibernate Search does use maps and a two-phase iteration approach, as an answer to user reported (real street) issues.

What I find more amusing is the unneccessary N loop introduced by using 

     CollectionHelper.initializeSizedList( searchObjects.length, null );

that's not premature optimisation, that's just pointless overhead.